parser.hpp

```diff
@@ -47,16 +47,14 @@ constexpr char encode_candy_type(CandyType t)
 
 constexpr CandyType decode_candy_type(char c)
 {
-    int i = 0;
-
-    for (; i < sizeof(map_candy_type); ++i) {
-        if (map_candy_type[i] == c) {
-            return static_cast<CandyType>(i);
-        }
-    }
-
-    if (i == sizeof(map_candy_type)) {
-        throw std::runtime_error("Invalid candy type");
+    switch (c)
+    {
+        case ' ': return CandyType::None;
```

Comment:
Nice, this will avoid a lot of unnecessary iterations.

Reply:
Inspired by Elm ;-)